pkg/pubsub/adapter/converters/pubsub_pull.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"fmt"
+	cev2 "github.com/cloudevents/sdk-go/v2"
+	. "github.com/cloudevents/sdk-go/v2/event"
+	"github.com/google/knative-gcp/pkg/apis/events/v1beta1"
+	. "github.com/google/knative-gcp/pkg/pubsub/adapter/context"
+)
+
+func convertPubSubPull(ctx context.Context, msg *pubsub.Message) (*cev2.Event, error) {
+	event := cev2.NewEvent(cev2.VersionV1)
+	event.SetID(msg.ID)
+	event.SetTime(msg.PublishTime)
+
+	project, err := GetProjectKey(ctx)
+	if err != nil {
+		return nil, err
+	}
+	topic, err := GetTopicKey(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	event.SetSource(v1beta1.CloudPubSubSourceEventSource(project, topic))
+	event.SetType(v1beta1.CloudPubSubSourcePublish)
```

Comment:
We can address this separately but I think this should be unset because we don't control the data in this case. Thoughts?

Reply:
this is for an internal opinionated Pullsubscription not the Channel. 
I think is fine to leave it...

Reply:
unless you think otherwise